index.d.ts

```diff
@@ -133,6 +133,22 @@ declare namespace ora {
 		readonly prefixText?: string | PrefixTextGenerator;
 	}
 
+	interface PromiseOptions<T> extends Options {
+		/**
+    The new text of the spinner when the promise is resolved.
+
+    If undefined, will keep the initial text.
+    */
+		successText?: string | ((result: T) => string);
+
+		/**
+    The new text of the spinner when the promise is rejected.
+
+    If undefined, will keep the initial text.
+    */
+		failText?: string | ((error: Error) => string);
```

Comment:
Still messy formatting. Please look over the whole diff.

Reply:
I don't know why this is happening at all. In my VSCode the indentation is right

Reply:
I didn't have EditorConfig extension installed here, I was using 2 spaces instead of tab (size 2). The code was looking the same in VSCode, but looks like the diff in GitHub renders tabs as 8 spaces, that's why it was visually strange.

I activated the whitespace rendering, and that's what was happening:

![image](https://user-images.githubusercontent.com/28177550/127792983-4f883223-d9cf-4695-aafc-2a24eb6830d0.png)

I found out that you are the XO dev. Maybe would be a good idea to have a pre-commit/GitHub action to lint this. Or even, to throw a warn/error on test if spaces were found instead of tabs. In eslint, this is doable.

I hope everything is fixed now.

Reply:
Just for your information, there is a cool eslint plugin for JSDoc: https://github.com/gajus/eslint-plugin-jsdoc

I will soon add it to my personal eslint-config, as I also have my personal taste for JSDoc comments. Maybe you could implement a similar implementation for XO to avoid those issues among all your repos, in PR like mine.

Reply:
> Or even, to throw a warn/error on test if spaces were found instead of tabs. In eslint, this is doable.

That's already a thing. Ora just didn't have the latest XO version.

However, you are still expected to look over your own PR diff.

Reply:
> Just for your information, there is a cool eslint plugin for JSDoc: gajus/eslint-plugin-jsdoc

I plan to add it at some point, but it's some work to evaluate all the rules, and I haven't had the time to do that yet: https://github.com/xojs/xo/issues/378